lib/AST/NameLookup.cpp

```diff
@@ -1958,8 +1984,8 @@ directReferencesForUnqualifiedTypeLookup(DeclNameRef name,
   auto lookup = evaluateOrDefault(ctx.evaluator,
                                   UnqualifiedLookupRequest{descriptor}, {});
   for (const auto &result : lookup.allResults()) {
-    if (auto typeDecl = dyn_cast<TypeDecl>(result.getValueDecl()))
-      results.push_back(typeDecl);
+    auto typeDecl = cast<TypeDecl>(result.getValueDecl());
+    results.push_back(typeDecl);
```

Comment:
Interesting! I'd like to understand this change. Was the old check not needed?

Reply:
Name lookup only returns TypeDecls if the TypesOnly flag is set.

Reply:
Thanks.